### DISCUSSION TITLE
Alternate Marketplace URL Not working with product.json

### DISCUSSION THREAD

**Discussion by balajib-b** [gitpod-io/openvscode-server#352] [General] [answered]
Does this issue occur when all extensions are disabled?: No

- VS Code Version: 1.67.1
- OS Version:  Ubuntu 20.04
Steps to Reproduce:

1. Change Marketplace URL to microsoft marketplace for testing
2. From this -  	"extensionsGallery": {
		"serviceUrl": "https://open-vsx.org/vscode/gallery",
		"itemUrl": "https://open-vsx.org/vscode/item",
		"resourceUrlTemplate": "https://open-vsx.org/vscode/asset/{publisher}/{name}/{version}/Microsoft.VisualStudio.Code.WebResources/{path}",
		"controlUrl": "",
		"recommendationsUrl": ""
	},

3. To this -  "extensionsGallery": {
      "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
    "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
    "itemUrl": "https://marketplace.visualstudio.com/items"
  },

4. Marketplace is still openvsx.org

**Comment by jeanp413** [comment, marked as answer]
@balajib-b you'll need to build from sources again with `yarn gulp vscode-reh-web-linux-x64-min` as product.json gets inlined for web resources

**Reply by balajib-b** [reply]
Okay @jeanp413, i will try it out and post here.

**Comment by guddu121** [comment]
Hello All,

I have this branch with me locally and trying to build from sources using  yarn gulp vscode-reh-web-linux-x64-min, I am using red hat linux and getting error that gulp is not found and am unable to install as well.

Is there any alternative to build the source code , looks like gulp is deprecated

**Reply by jeanp413** [reply]
@guddu121 mayeb you are missing some dependencies :thinking: Check https://github.com/microsoft/vscode/wiki/How-to-Contribute

**Reply by guddu121** [reply]
Hi,

I am dockerzing the application and using our private image of linux, Even yarn was not availabe to so download, i tried to download gulp and i can see it is adding but when running executing the command "yarn gulp vscode-linux-x64" it throws error.

I am unable to figure our what is missing as i am getting warning that gulp is deprecated. Is there any alternative way to build the source.

What i am trying to achive is 
1> Trying to configure our prive dns to run vscode
2> Trying to add extension to be able to use in vs code
i need to build the source to be able to test the above

**Reply by guddu121** [reply]
aslo, want to know how to get the name "vscode-reh-web-linux-x64-min" for mine setup. I m unsure if this shold be same for me as well

**Reply by jeanp413** [reply]
> I am dockerzing the application and using our private image of linux, Even yarn was not availabe to so download, i tried to download gulp and i can see it is adding but when running executing the command "yarn gulp vscode-linux-x64" it throws error.

assuming your image linux distro is a [supported platform](https://code.visualstudio.com/docs/remote/linux) you can just build the server on ubuntu and it should work on your private image

> aslo, want to know how to get the name "vscode-reh-web-linux-x64-min" for mine setup.

if your arch is other than `x64` then you'll need to change that accordingly, check `build/gulpfile.reh.js` file

**Reply by guddu121** [reply]
Hello all,

Help need to deploy openvsx code server locally.
I am trying to build the solution using docker file so i have downloaded the code from the tag version 1.73.4
Within docker file i am doing below things
RUN npm config set regisry "<our registry>"
RUN npm set strict-ssl false
RUN npm install -g gulp-cli
RUN npm install -g gulp
RUN npm install yarn --global &&
yarn config set registry "registry_url"
&&
yarn config set "strict-ssl" false -g
RUN yarn install
RUN yarn gulp vscode-reh-web-linux-x64-min

The above code coming out with error when i am commenting out "RUN yarn gulp vscode-reh-web-linux-x64-min" , error "@policy-watcher package not found and number of error relatd to python packages

when i exclude yarn install then i get error that gulp.js is not found under node_module directory
#RUN yarn install
RUN yarn gulp vscode-reh-web-linux-x64-min

Can you please shed a light that how to fix and build the souce code using "RUN yarn gulp vscode-reh-web-linux-x64-min"

I can install gulp locally using  "npm install gulp " from docker file. Using gloabl key word i can install gulp but it do not shows gulp local version it just shows gulp cli version
